### ISSUE TITLE
gr.spatial_neighbors fails if Delaunay returns isolated nodes

### ISSUE THREAD

**Issue by marcovarrone** [theislab/squidpy#386]
### Description
I notice that `scipy.spatial.Delaunay` may not assign certain nodes to any simplex.
With the current implementation the function `gr.spatial_neighbors` fails because it passes an empty list to the `euclidean_distances` function

### Minimal reproducible example
Unfortunately, I have not been able to reproduce the problem with a simple example.
I cannot understand in which conditions a node is not assigned in any simplex.
I tried to set up a dataset with a group of few nodes close to each other and one node very far from that group but the latter is still connected to some of the others.

### Traceback

<details>

```pytb
~/miniconda3/envs/squidpy/lib/python3.8/site-packages/squidpy/gr/_build.py in spatial_neighbors(adata, spatial_key, coord_type, n_rings, n_neigh, delaunay, radius, transform, key_added)
    115 
    116     elif coord_type == CoordType.GENERIC:
--> 117         Adj, Dst = _build_connectivity(coords, n_neigh, radius, delaunay=delaunay, return_distance=True)
    118     else:
    119         raise NotImplementedError(coord_type)

~/miniconda3/envs/squidpy/lib/python3.8/site-packages/squidpy/gr/_build.py in _build_connectivity(coords, n_neigh, radius, delaunay, neigh_correct, set_diag, return_distance)
    172 
    173                 #if len(vertex_neighs) > 0:
--> 174                 dists_neighs = euclidean_distances(
    175                     coords[vertex_neighs, :], coords[np.newaxis, i, :]
    176                 )

~/miniconda3/envs/squidpy/lib/python3.8/site-packages/sklearn/utils/validation.py in inner_f(*args, **kwargs)
     61             extra_args = len(args) - len(all_args)
     62             if extra_args <= 0:
---> 63                 return f(*args, **kwargs)
     64 
     65             # extra_args > 0

~/miniconda3/envs/squidpy/lib/python3.8/site-packages/sklearn/metrics/pairwise.py in euclidean_distances(X, Y, Y_norm_squared, squared, X_norm_squared)
    270            [1.41421356]])
    271     """
--> 272     X, Y = check_pairwise_arrays(X, Y)
    273 
    274     # If norms are passed as float32, they are unused. If arrays are passed as

~/miniconda3/envs/squidpy/lib/python3.8/site-packages/sklearn/utils/validation.py in inner_f(*args, **kwargs)
     61             extra_args = len(args) - len(all_args)
     62             if extra_args <= 0:
---> 63                 return f(*args, **kwargs)
     64 
     65             # extra_args > 0

~/miniconda3/envs/squidpy/lib/python3.8/site-packages/sklearn/metrics/pairwise.py in check_pairwise_arrays(X, Y, precomputed, dtype, accept_sparse, force_all_finite, copy)
    144                             estimator=estimator)
    145     else:
--> 146         X = check_array(X, accept_sparse=accept_sparse, dtype=dtype,
    147                         copy=copy, force_all_finite=force_all_finite,
    148                         estimator=estimator)

~/miniconda3/envs/squidpy/lib/python3.8/site-packages/sklearn/utils/validation.py in inner_f(*args, **kwargs)
     61             extra_args = len(args) - len(all_args)
     62             if extra_args <= 0:
---> 63                 return f(*args, **kwargs)
     64 
     65             # extra_args > 0

~/miniconda3/envs/squidpy/lib/python3.8/site-packages/sklearn/utils/validation.py in check_array(array, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, estimator)
    667         n_samples = _num_samples(array)
    668         if n_samples < ensure_min_samples:
--> 669             raise ValueError("Found array with %d sample(s) (shape=%s) while a"
    670                              " minimum of %d is required%s."
    671                              % (n_samples, array.shape, ensure_min_samples,

ValueError: Found array with 0 sample(s) (shape=(0, 3)) while a minimum of 1 is required by check_pairwise_arrays.
```

</details>

### Version
1.1.0

The fix is quite simple so I am already working on a PR :)

**Comment by michalk8** [comment]
Hi @marcovarrone , thanks for catching this, looking forward to the PR!

**Comment by michalk8** [comment]
closed via #393
